src/rewrite_reduce.cpp

```diff
@@ -0,0 +1,38 @@
+#include <migraphx/rewrite_reduce.hpp>
+#include <migraphx/module.hpp>
+#include <migraphx/matcher.hpp>
+#include <migraphx/make_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+namespace {
+struct find_softmax
+{
+    auto matcher() const { return match::name("softmax"); }
+
+    void apply(module& m, const match::matcher_result& r) const
```

Comment:
Does this make softmax JIT kernel redundant ?

Reply:
Yes it will.

Reply:
This would make matching flash attention more complicated, which is why I kept it. We could consider removing this in the future.

Also, we might want to calibrate this operator in the future so we can get rid of the reduce_max.